src/helpers/ApplicationPropertiesFinder.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import vscode from 'vscode';
+import path from 'path';
+import { Suggestion } from './SuggestionRegistry';
+
+export async function findAllApplicationPropertiesFiles(startUri: vscode.Uri): Promise<vscode.Uri[]> {
+	let currentDir = path.dirname(startUri.fsPath);
+	const workspaceFolder = vscode.workspace.getWorkspaceFolder(startUri)?.uri.fsPath;
+
+	while (workspaceFolder && currentDir.startsWith(workspaceFolder)) {
+		const entries = await vscode.workspace.fs.readDirectory(vscode.Uri.file(currentDir));
+		const matchingFiles = entries
+			.filter(([name, type]) => {
+				return type === vscode.FileType.File && /^application(-.+)?\.properties$/.test(name);
```

Comment:
Creating RegExp objects is expensive in JavaScript, and considering we need to run this function several times when working with the configuration form, could we explore a slightly different mechanism to filter the files?

I'm wondering if this check could work:
```typescript
return type === vscode.FileType.File && name.startsWith('application') && name.endsWith('.properties');
```

Alternatively, we could create the RegExp once, outside of the function, and then use it and reset it immediately after:
```typescript
// Defined outside the function
const APP_PROPERTIES_REGEXP = /^application(-.+)?\.properties$/;

// Inside of the function
APP_PROPERTIES_REGEXP.lastIndex = 0;
return type === vscode.FileType.File && APP_PROPERTIES_REGEXP.test(name);
```